loaders/embededKeyLoader_test.go

```diff
@@ -0,0 +1,314 @@
+		require.Error(t, err)
+	})
+
+	t.Run("no cache", func(t *testing.T) {
+		mockLoader := &MockKeyLoader{
+			keys: map[circuits.CircuitID][]byte{
+				testID: testKey,
+			},
+		}
+		loader := NewEmbeddedKeyLoader(
+			WithKeyLoader(mockLoader),
+			WithCacheDisabled(),
+		)
+
+		key, err := loader.Load(testID)
+		require.NoError(t, err)
+		assert.Equal(t, testKey, key)
+
+		// Verify key was not cached
+		assert.Nil(t, loader.cache)
+	})
+
+	t.Run("embedded key not found", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+
+		_, err := loader.Load("non-existent-circuit")
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "failed to load default key")
+	})
+}
+
+func TestDefaultEmbeddedKeys(t *testing.T) {
+
+	t.Run("authV2", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AuthV2CircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("AtomicQueryMTPV2CircuitID", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AtomicQueryMTPV2CircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("AtomicQueryMTPV2OnChainCircuitID", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AtomicQueryMTPV2OnChainCircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("AtomicQuerySigV2CircuitID", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AtomicQuerySigV2CircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("AtomicQuerySigV2CircuitID", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AtomicQuerySigV2OnChainCircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("AtomicQueryV3CircuitID - beta.1", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AtomicQueryV3CircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("AtomicQueryV3OnChainCircuitID - beta.1", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AtomicQueryV3OnChainCircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("LinkedMultiQuery10CircuitID - beta.1", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.LinkedMultiQuery10CircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+}
+
+func TestEmbeddedKeyLoader_Load_EmbeddedKeys(t *testing.T) {
+	t.Run("load embedded key", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		key, err := loader.Load(circuits.AuthV2CircuitID)
+		require.NoError(t, err)
+		assert.NotNil(t, key)
+	})
+
+	t.Run("embedded key not found", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		_, err := loader.Load("non-existent-circuit")
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "failed to load default key")
```

Comment:
Compare error text is bad practice. Try to use assert.ErrorsIs or assert.ErrorAs

Reply:
fix